_posts/2023-03-07-nextjs-at-artsy-retrospective.markdown

```diff
@@ -0,0 +1,375 @@
+---
+layout: epic
+title: "Two years of Next.js at Artsy: A Retrospective"
+subtitle: "TODO"
+date: 2024-03-07
+categories: [Next.js, React]
+author: [christopher-pappas]
+comment_id: TODO
+---
+
+Where to begin! Where to begin...
+
+Lets start with a good measure of success, one that I think most engineers could
+agree with: a from-scratch rebuild of a complex, internal CMS. Success with
+rebuilds is often fleeting, as many can attest. There's risk involved. Extending
+the challenge further, lets assign (most) of the task to a team of DevOps
+engineers who aren't too familiar with modern front-end technology, JavaScript
+tooling, web frameworks, and more. And then say that we succeeded -- like wild
+success.
+
+Is Artsy crazy? Or is Next.js (along with our team :) simply that good? Well,
+it's a little of both. And it's a little bittersweet, in a way. Here's our tale.
+
+<!-- more -->
+
+### Beginnings
+
+Our first formal use of Next came from a project spun up during Hackathon, two
+years ago. Hackathon's are great opportunities to push technology forward, and
+Artsy has had a lot of success with them. We'd been talking about Next.js for a
+while, and experimenting here and there, but we were never able to find the
+proper intersection of product and purpose to take things further. With this in
+mind, one of our Engineers (Roop :wave:) spun up a Next.js POC that allowed us
+to deduplicate artist records. On the surface, the effort was non-trivial. It
+involved authentication, DB communication, and more. Yet when presented during
+Hackathon, all of this was done and it also looked _beautiful_.
+
+How could so much product work get completed in such a short period of time? A
+large part of that is due to Next.js's framework design, and how it simply gets
+out of the way and allows one to start building product features. The defaults
+made sense; to create a new route, simply create a new folder or file in the
+`pages` directory, and Next takes care of the rest. And what about compilation,
+TypeScript, linting, and all of the other extraordinarily confusing JavaScript
+toolchain details that folks typically struggle over? All taken care of, at the
+framework level, as if they didn't even exist at all. The API side of things was
+light and easy to understand, thanks to great documentation.
+
+There are omitted details of course, and there were things we needed to figure
+out (such as auth, via `next-auth`), but in any case this gave us a lot of
+confidence to start discussing what it might take to seriously consider
+rebuilding our internal tools app. Torque, as it was known, was an old homegrown
+Node.js framework built on top of Express, Backbone and CoffeeScript. A few
+meetings later it was decided, and our platform team agreed to take it (with
+loose backup support from a couple other engineers). And a short few month's
+later it was pretty much complete, executed by a team working amidst unfamiliar
```

Comment:
I might soften "pretty much complete" due to the long tail of things that still stubbornly live in Torque, but it's true there was a lot of progress!